From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Fri, 15 Feb 2019 13:33:52 +0100
Subject: atk key: Always convert control character events into key names

including when the control modifier is present, i.e. when one is typing
control-I for instance.

Orca would convert them back to the corresponding ASCII letter anyway, and
when pressing control-tab, we do want to pass "tab", not pass "\t" that Orca
would erroneously convert to "control-I".

Fixes #1743

(cherry picked from commit 728f6869cba6360b843a83fe5b525404accb1433)
Origin: upstream, 2.24.33, commit:cf15c11429559a180eb1b185da02b348211d860c
---
 modules/other/gail/gailutil.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/modules/other/gail/gailutil.c b/modules/other/gail/gailutil.c
index cab0113..ab97c77 100644
--- a/modules/other/gail/gailutil.c
+++ b/modules/other/gail/gailutil.c
@@ -224,8 +224,7 @@ atk_key_event_from_gdk_event_key (GdkEventKey *key)
   event->keyval = key->keyval;
   event->length = key->length;
   if (key->string && key->string [0] && 
-      (key->state & GDK_CONTROL_MASK ||
-       g_unichar_isgraph (g_utf8_get_char (key->string))))
+      g_unichar_isgraph (g_utf8_get_char (key->string)))
     {
       event->string = key->string;
     }
